src/Dialect/ONNX/ONNXOps/Math/ElementwiseBroadcast.cpp

```diff
@@ -28,10 +28,10 @@ using namespace onnx_mlir;
 namespace {
 
 // Verify shape for numpy style broadcasting operators.
-template <class OP, class ADAPTOR>
+template <class OP_TYPE>
 static LogicalResult verifyShapeForBroadcastingOps(
-    OP &op, Type elementType = nullptr) {
-  ADAPTOR operandAdaptor(op);
+    OP_TYPE &op, Type elementType = nullptr) {
```

Comment:
Could we avoid template by using `Operation *op` instead, and move it to a common place, e.g. `ShapeHelper.cpp`? I want to use this function in NNPA code also. It seems `operandAdator` is used to get operands that can be obtained from `op->getOperands()`.

Reply:
did it in the latest version